src/main/java/com/redhat/pantheon/servlet/ModuleListingServlet.java

```diff
@@ -51,19 +53,23 @@ protected String getQuery(SlingHttpServletRequest request) {
         // FIXME Searching by resourceType because in some cases, searching directly on the primaryType
         // is not returning any results
         StringBuilder queryBuilder = new StringBuilder()
-                .append("select m.* from [nt:base] as m ")
-                    .append("INNER JOIN [pant:moduleVersion] as rev ON ISDESCENDANTNODE(rev, m) ")
-                .append("where (isdescendantnode(m, '/content/repositories') ")
-                    .append("or isdescendantnode(m, '/content/modules') ")
-                    .append("or isdescendantnode(m, '/content/sandbox')) ")
-                .append("AND m.[jcr:primaryType] = 'pant:module' ")
-                // look in ALL versions (all locales)
-                .append("AND (rev.[metadata/jcr:title] like '%" + searchParam + "%' ")
-                    .append("OR rev.[metadata/jcr:description] like " + "'%" + searchParam + "%') ");
+                .append("SELECT m.* from [nt:base] AS m ")
+                .append("LEFT OUTER JOIN [nt:base] AS loc ON  ISCHILDNODE(loc, m) ")
+                .append("LEFT OUTER JOIN [nt:base] AS draft ON  draft.[jcr:uuid] = loc.[draft] ")
+                .append("LEFT OUTER JOIN [nt:base] AS release ON  release.[jcr:uuid] = loc.[released] ")
+                .append("WHERE m.[jcr:primaryType] = 'pant:module' ")
+                .append("AND loc.[jcr:primaryType] = 'pant:moduleLocale' ")
+                .append("AND (draft.[jcr:primaryType] = 'pant:moduleVersion' OR draft.[jcr:primaryType] IS NULL) ")
+                .append("AND (release.[jcr:primaryType] = 'pant:moduleVersion' OR release.[jcr:primaryType] IS NULL) ")
+                .append("AND (draft.[metadata/jcr:title] LIKE '%" + searchParam + "%' ")
+                    .append("OR draft.[metadata/jcr:description] LIKE '%" + searchParam + "%' ")
+                    .append("OR release.[metadata/jcr:title] LIKE '%" + searchParam + "%' ")
+                    .append("OR release.[metadata/jcr:description] LIKE '%" + searchParam + "%')");
 
         if(!isNullOrEmpty(keyParam) && !isNullOrEmpty(directionParam)) {
-            queryBuilder.append(" order by rev.[metadata/")
-                    .append(keyParam).append("] ")
+            queryBuilder.append(" ORDER BY coalesce(release.[metadata/")
+                    .append(keyParam).append("],draft.[metadata/")
```

Comment:
Hey Ankit, please reverse the order between `release` and `draft` here. We need to have `draft` as the first parameter for `coalesce`, followed by `release`. If you have `release` first, you get weird behavior when sorting by Upload Time - where the thing that you just uploaded a draft version for doesn't necessarily get sorted to the top of the list. I mode the code update on my local system and it appears to work properly. Let me know if you disagree with the change. Otherwise this looks great. Thanks!

Reply:
Hey Ben, 
Thanks for pointing this out.
I have made the changes and tested and it looks good to me now.
Let me know if there are any more issues.